### ISSUE TITLE
Docker Jenkins Build Collector

### ISSUE THREAD

**Issue by no1melman** [Hygieia/Hygieia#3014]
I've set up the jenkins build collector with these ENV values:

```
  jenkins-build-collector:
    environment:
     - MONGODB_PORT=27017
     - MONGODB_HOST=db
     - MONGO_PORT=27017
     - MONGO_HOST=db
     - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_DATABASE=dashboarddb
     - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_USERNAME=dashboarduser
     - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_PASSWORD=dbpassword

```
This fails at start up as the mongo can't connect:

```
2019-06-28 13:42:44,115 INFO  org.mongodb.driver.cluster - Exception in monitor thread while connecting to server localhost:27017
com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.SocketStream.open(SocketStream.java:63) ~[mongo-java-driver-3.0.2.jar!/:na]
	at com.mongodb.connection.InternalStreamConnection.open(InternalStreamConnection.java:114) ~[mongo-java-driver-3.0.2.jar!/:na]
	at com.mongodb.connection.DefaultServerMonitor$ServerMonitorRun
```

As you can see, it is actually using `localhost:27017` to connect - which to me makes no sense... I've even gone into the `properties-builder.sh` and set it there.... still no good...

For comparison, I've setup the github collector with the same ENV values and it works:

```
  github-collector:
    environment:
      - MONGODB_PORT=27017
      - MONGODB_HOST=db
      - MONGO_PORT=27017
      - MONGO_HOST=db
      - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_DATABASE=dashboarddb
      - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_USERNAME=dashboarduser
      - HYGIEIA_API_ENV_SPRING_DATA_MONGODB_PASSWORD=dbpassword
```

**Comment by rvema** [comment]
@no1melman let me try recreating it

**Comment by no1melman** [comment]
@rvema I've forced the dbhost name in the hygieia core, then got a new error:

```
Caused by: java.lang.NullPointerException
	at org.springframework.scheduling.support.CronSequenceGenerator.parse(CronSequenceGenerator.java:265)
	at org.springframework.scheduling.support.CronSequenceGenerator.<init>(CronSequenceGenerator.java:95)
	at org.springframework.scheduling.support.CronSequenceGenerator.<init>(CronSequenceGenerator.java:82)
	at org.springframework.scheduling.support.CronTrigger.<init>(CronTrigger.java:44)
	at com.capitalone.dashboard.collector.CollectorTask.onStartup(CollectorTask.java:73)
```

From what I understand, the HudsonCollectorTask implements Collector. Collector has an abstract method `getCron()`. The point of this is to allow the HudsonCollectorTask to read its HudsonSettings to get the cron... So to me, it seems these settings aren't being read. I checked up the HudsonSettings and it doesn't have a attribute to read a particular property... 

I'm don't know how spring boot gets its properties or where it is expecting to get the value from - but it appears to be null.... 

This is kind of suspicious seeing as the host variable was also ignored until set in the `@Value` attribute (I assume this is used to assign a default if can't be found).

So it appears that properties aren't being read properly? Where would these Settings classes be looking for their values?
